### ISSUE TITLE
wistia intermittently fails to embed video 

### ISSUE THREAD

**Issue by lkraav** [wistia/wistia-wordpress-plugin#12]
I have a site where nothing happens when just pasting the URL in.

When surrounding the video with `[embed]...[/embed]` I get a progress bar and then a few seconds later an error message "failed to embed".

Nothing visible in `debug.log`. It feels like a bug in the plugin.

**Comment by lkraav** [comment]
Is the oembed regex capable of properly handling `&` vs `&amp;` and themsuch?

**Comment by MaxPower15** [comment]
Hi @lkraav. When you pasted the URL, did you paste it on its own line, with blank lines both before and after it? The oembed code is powered almost entirely by WordPress (we just give it an endpoint to hit), so if that's not working, it's likely turned off or being overridden by your theme.

If the URL somehow made it through with html escaped ampersands, then I would still expect the oembed endpoint to return some kind of an embed code. As the oembed endpoint doesn't make assumptions about the kind of strings it will get, this should work, unless the URL that WordPress sends to us is malformed.

I am not sure how oEmbed URLs would interact with `[embed]` tags--that's not something we've tested with this plugin.

As for something to try, I would temporarily (or in a local environment) switching themes or disabling other plugins that may be causing trouble. That would help you narrow down the cause of the issue.

**Comment by lkraav** [comment]
Heya. I've established that yes, it's definitely something to do with intereference from another plugin, but it's really hard to tell which one. I've done a bunch of deactivation testing, but failed to establish any clear pattern. The embed does work on a clean install.

Will have to get to the bottom of this sooner or later, and I'll update the ticket then.

**Comment by lkraav** [comment]
I never knew the plugin had the option page and ability to toggle this "legacy mangler" thing. Maybe it has been the source of all my worries? Will check and compare the next time the videos break.

**Comment by MaxPower15** [comment]
@lkraav Yeah, give it a try and let me know. I just added that settings page a couple days ago. The Anti-Mangler was created with good intentions but once oEmbeds became really solid, it was mostly just a lot of code slowly getting more and more out of date, and messing things up in the worst case. I'm glad to turn it off by default!

**Comment by lkraav** [comment]
I'm fairly certain I've narrowed it down to conflict with https://github.com/ahoereth/featured-video-plus cc @ahoereth

As soon as FVP is disabled, everything works.

When FVP is enabled, any oembed that has the `embedType` parameter will fail to embed. @ahoereth your thoughts?
